core/extrinsics_submission_service/extrinsic_submission_api.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_EXTRINSIC_SUBMISSION_API_HPP
+#define KAGOME_CORE_EXTRINSICS_SUBMISSION_SERVICE_EXTRINSIC_SUBMISSION_API_HPP
+
+/**
+ * ExtrinsicSubmissionApi based on auth api implemented in substrate here
+ * https://github.com/paritytech/substrate/blob/e8739300ae3f7f2e7b72f64668573275f2806ea5/core/rpc/src/author/mod.rs#L50-L49
+ */
+
+#include <boost/variant.hpp>
+#include <outcome/outcome.hpp>
+#include "common/visitor.hpp"
+#include "crypto/hasher.hpp"
+#include "extrinsics_submission_service/error.hpp"
+#include "primitives/auth_api.hpp"
+#include "primitives/extrinsic.hpp"
+
+namespace kagome::transaction_pool {
+  class TransactionPool;
+}
+
+namespace kagome::runtime {
+  class TaggedTransactionQueue;
+}
+
+namespace kagome::service {
+  class ExtrinsicSubmissionApi {
+    template <class T>
+    using sptr = std::shared_ptr<T>;
+
+   public:
+    /**
+     * @constructor
+     * @param api ttq instance shared ptr
+     * @param pool transaction pool instance shared ptr
+     * @param hasher hasher instance shared ptr
+     */
+    ExtrinsicSubmissionApi(
+        std::shared_ptr<runtime::TaggedTransactionQueue> api,
+        std::shared_ptr<transaction_pool::TransactionPool> pool,
+        std::shared_ptr<hash::Hasher> hasher);
+
+    /**
+     * @brief validates and sends extrinsic to transaction pool
+     * @param bytes encoded extrinsic
+     * @return hash of successfully validated extrinsic
+     * or error if state is invalid or unknown
+     */
+    outcome::result<common::Hash256> submit_extrinsic(
+        const common::Buffer &bytes);
+
+    /**
+     * @return collection of pending extrinsics
+     */
+    outcome::result<std::vector<std::vector<uint8_t>>> pending_extrinsics();
```

Comment:
it is bad that extrinsic here is just a std::vector<uint8_t>

Can we use a type here?
Perhaps
```
struct Extrinsic{
  std::vector<uint8_t> data;
}
```

Reply:
sounds reasonable here.

Reply:
done

Reply:
I see now what you mean, I used primitives::Extrinsic